admin-auth/src/main/kotlin/com/epam/drill/admin/auth/config/OAuthConfig.kt

```diff
@@ -31,40 +34,82 @@ class OAuthConfig(private val config: ApplicationConfig) {
             .config("auth")
             .config("oauth2")
 
+    /**
+     * An OAuth2 authorize URL.
+     */
     val authorizeUrl: String
         get() = oauth2.property("authorizeUrl").getString()
 
+    /**
+     * An OAuth2 access token URL.
+     */
     val accessTokenUrl: String
         get() = oauth2.property("accessTokenUrl").getString()
 
+    /**
+     * An OAuth2 user info URL. Optional, if not set, the request for user information will not be used.
+     */
     val userInfoUrl: String?
         get() = oauth2.propertyOrNull("userInfoUrl")?.getString()
 
+    /**
+     * A OAuth2 client ID.
+      */
     val clientId: String
         get() = oauth2.property("clientId").getString()
 
+    /**
+     * A OAuth2 client secret.
+     */
     val clientSecret: String
         get() = oauth2.property("clientSecret").getString()
 
+    /**
+     * Scopes to OAuth2 request. Optional, empty list by default.
+     */
     val scopes: List<String>
         get() = oauth2.propertyOrNull("scopes")?.getString()
             ?.split(",")?.map { it.trim() }?.filter { it.isNotEmpty() }
             ?: listOf()
 
+    /**
+     * UI application root URL. The key is "drill.ui.rootUrl".
+     */
     val uiRootUrl: String
         get() =  ui.property("rootUrl").getString()
 
+    /**
+     * UI application root path. The key is "drill.ui.rootPath". Optional, "/" by default.
+     */
     val uiRootPath: String
         get() = ui.propertyOrNull("rootPath")?.getString() ?: "/"
 
+    /**
+     * OAuth2 access token payload mapping.
+     *
+     * Used for map access token claims to [com.epam.drill.admin.auth.model.UserInfoView].
+     * Includes username and roles mapping:
+     * A username mapping is a name of field containing the username field in the access token payload. Optional, "sub" by default.
+     * A roles mapping is a name of field containing roles in access token payload. Optional, if not set, roles are not mapped.
+     *
+     * @see [Standard claims](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims)
+     */
     val tokenMapping: UserMapping
         get() = oauth2.config("tokenMapping").run {
             UserMapping(
-                username = propertyOrNull("username")?.getString() ?: "sub", //see https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims
+                username = propertyOrNull("username")?.getString() ?: "sub",
                 roles = propertyOrNull("roles")?.getString()
             )
         }
 
+    /**
+     * OAuth2 user info mapping.
```

Comment:
-> `OAuth2 /user-info mapping`
State precedence over token payload, e.g.:
`If /user-info mapping is specified values extracted from token [tokenMapping] are ignored (values mapped from /user-info take precedence)`

Reply:
Explain that `userInfoUrl` is the key that defines